tests/e2e/matmul/CMakeLists.txt

```diff
@@ -2222,6 +2222,39 @@ iree_generated_e2e_runner_test(
     "requires-gpu-cdna3"
 )
 
+elseif(IREE_HIP_TEST_TARGET_CHIP MATCHES "^gfx94")
+
+# I8 Intrinsics has different layout on CDNA3/gfx94x,
+# and only CDNA3/gfx94x has F8 intrinsics.
+
+iree_generated_e2e_runner_test(
+  NAME
+    e2e_matmul_rocm_f8_large_cdna3_mfma
+  TEST_TYPE
+    matmul
+  GENERATOR
+    "generate_e2e_matmul_tests.py"
+  GENERATOR_ARGS
+    "--lhs_rhs_type=f8E4M3FNUZ"
+    "--acc_type=f32"
+    "--shapes=gpu_large_aligned"
+    "--compilation_info=LLVMGPUVectorDistributeMFMA"
```

Comment:
Can these intrinsics and the compilation info be inferred from the target chip? I'd like to avoid more branches in test files.

At the very least this should have an explanation for why a new branch is being added in the PR description.

Reply:
> Can these intrinsics and the compilation info be inferred from the target chip? I'd like to avoid more branches in test files.

Yes, but I am not sure how to use that information to stop runner without those features to compile it.  Since if it tries to compile with that intrinsic, it will have compilation failure.

> At the very least this should have an explanation for why a new branch is being added in the PR description.

The main reasons are:
1. Compilation with target chip that do not have the FP8 intrinsics would fail in iree-compile. 
2. I8 Intrinsics has different layout on CDNA3/gfx94x, and only CDNA3/gfx94x has F8 intrinsics.

I can edit the PR to have these if you think it suffice :)